Util/channel/src/main/java/io/deephaven/util/channel/SeekableChannelsProvider.java

```diff
@@ -85,4 +86,24 @@ default SeekableByteChannel getWriteChannel(@NotNull final String path, final bo
     }
 
     SeekableByteChannel getWriteChannel(@NotNull Path path, boolean append) throws IOException;
+
+    /**
+     * Applies the supplied action on the list of file URIs contained inside the given directory. The logic is
+     * non-recursive.
+     *
+     * @param directoryURI The URI of the directory to list
+     * @param processor A method to apply to the URIs in the directory
+     */
+    void applyToChildURIs(@NotNull URI directoryURI, @NotNull Consumer<URI> processor) throws IOException;
```

Comment:
I don't love the naming. I also don't like the pattern - that is, passing down a consumer as opposed to returning something that can be iterated over. I think:

```java
Stream<URI> list(URI directory);

Stream<URI> walk(URI directory);
```

makes it easier from a consumer point of view. The FS-based impl will be very easy to implement. The S3 ones will be a little trickier, but nothing too complex - should be able to use an Iterator-based impl. Happy to help. Thoughts?